### PR TITLE
Create No-Privilege-Escalation PSP policy

### PR DESCRIPTION
**Pull request by EmandM**

### CODE REVIEW

File: built-in-policies/policyDefinitions/Kubernetes/ContainerNoPrivilegeEscalation_EnforceOPAConstraint.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+  "properties": {
```
Comment (pilor) [inline built-in-policies/policyDefinitions/Kubernetes/ContainerNoPrivilegeEscalation_EnforceOPAConstraint.json:2]: Built-in policies are only checked into this repo by an automated system each time we release updates to Azure.  They live in the AzureUX-ARM repo internally.  Contact me (cheggert) if you need a pointer to their location.

Comment (EmandM) [reply built-in-policies/policyDefinitions/Kubernetes/ContainerNoPrivilegeEscalation_EnforceOPAConstraint.json]: Sweet, I've removed this file, will coordinate with Ramya for committing the policy definitions

File: built-in-references/Kubernetes/container-no-privilege-escalation/template.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+metadata:
+  name: k8sazurecontainernoprivilegeescalation
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sAzureContainerNoPrivilegeEscalation
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
```
Comment (RamyasreeChakka) [inline built-in-references/Kubernetes/container-no-privilege-escalation/template.yaml:12]: How is this policy different from the existing policy https://github.com/Azure/azure-policy/blob/master/built-in-references/Kubernetes/container-no-privilege/template.yaml? Do we need another policy?

File: built-in-references/Kubernetes/container-no-privilege-escalation/constraint.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+apiVersion: constraints.gatekeeper.sh/v1beta1
```
Comment (RamyasreeChakka) [inline built-in-references/Kubernetes/container-no-privilege-escalation/constraint.yaml:1]: Before pushing policies into built-in-references folder here, similar to policies present in https://github.com/Azure/azure-policy-kubernetes/tree/prod/policies/GatekeeperV3, can you please write src.rego, src_test.rego to help understand the test coverage? Please check-in the policies along with tests first to azure-policy-Kubernetes repo or RegoPolicy repo.

File: built-in-references/Kubernetes/container-no-privilege-escalation/template.yaml (hunk 2 above)
Comment (EmandM) [reply built-in-references/Kubernetes/container-no-privilege-escalation/template.yaml:12]: Privilege escalation is different from a privileged container. It is possible to have an unprivileged container that has permissions to escalate those privileges

File: built-in-references/Kubernetes/container-no-privilege-escalation/constraint.yaml (hunk 3 above)
Comment (EmandM) [reply built-in-references/Kubernetes/container-no-privilege-escalation/constraint.yaml:1]: Sure, will push those up now then

**Comment by RamyasreeChakka** [comment]
Closing pull request as the policies with tests are added in community-policy repo.

**Pull request closed without merging**